resources/aws/gateway.go

```diff
@@ -103,15 +88,18 @@ func (g *Gateway) CreateOrFail() error {
 }
 
 func (g *Gateway) Delete() error {
-	gateway, err := g.findExisting()
+	gateways, err := g.list()
 	if err != nil {
 		return microerror.MaskAny(err)
 	}
 
-	if _, err := g.Clients.EC2.DeleteInternetGateway(&ec2.DeleteInternetGatewayInput{
-		InternetGatewayId: gateway.InternetGatewayId,
-	}); err != nil {
-		return microerror.MaskAny(err)
+	for _, gateway := range gateways {
+		_, err := g.Clients.EC2.DeleteInternetGateway(&ec2.DeleteInternetGatewayInput{
+			InternetGatewayId: gateway.InternetGatewayId,
+		})
+		if err != nil {
+			return microerror.MaskAny(err)
+		}
 	}
```

Comment:
Agree for informing the user about the resource not existing (which we are not doing atm). It would be nice to do it with a type assertion instead of a string comparison though. Cfr #112.